### PR TITLE
Filter sort pin

### PR DESCRIPTION
**Pull request by AtulPant2704**
- Sort Notes using Priority and Date
- Pin Note at the start

### CODE REVIEW

File: src/utils/noteHandlers/pinNoteHandler.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    const response = await pinNoteService(token, note);
+    if (response.status === 200) {
+      notesDispatch({ type: "EDIT_NOTE", payload: response.data.notes });
+      toast.success("Note pinned successfully");
+    } else {
+      throw new Error();
+    }
+  } catch (error) {
+    console.log(error);
+    // toast.error(error.response.data.errors[0]);
```
Comment (AnkurChunekar) [inline src/utils/noteHandlers/pinNoteHandler.js]: Remove this if not needed

File: src/components/Note/Note.jsx [hunk 2]
```diff
@@ -102,15 +115,15 @@ const Note = ({ note, setShowNoteModal, setEditNote }) => {
...
           )}
-          {pathname !== "/archive" && pathname !== "/trash" ? (
+          {/* {pathname !== "/archive" && pathname !== "/trash" ? (
             <button
               title="Edit"
               className="action-btn"
               onClick={editNoteHandler}
             >
               <span className="material-icons-outlined nav-icon">edit</span>
             </button>
```
Comment (kart-c) [inline src/components/Note/Note.jsx]: you can remove these comments if not required

**Review by kart-c — APPROVED** [review]
- You can change the date format to `dd-mm-yyyy`.
- You can close filter modal/component on clicking outside/body
- The UI looks good

**Comment by AnkurChunekar** [comment]
Bug: If I add more than one note within a minute and try to sort the notes by time, the sorted notes are rendered in reverse order, please look into this. Also this can be resolved if you provide proper date string including `ms`.

**Comment by AnkurChunekar** [comment]
Reference: https://www.awesomescreenshot.com/video/8737551?key=36d17f4f952fdab916ad49700428cc98
After viewing a note if I try to add a new note, the add note modal displays all the data of previously viewed note.

**Review by AnkurChunekar — APPROVED** [review]
Other than the changes suggested everything looks good.
- Code is neat and clean.
- UI looks good.
- Good File Structure.

**Comment by AtulPant2704** [comment]
> Bug: If I add more than one note within a minute and try to sort the notes by time, the sorted notes are rendered in reverse order, please look into this. Also this can be resolved if you provide proper date string including `ms`.

Thanks for pointing it out, it's fixed now.

**Pull request merged**